lib/path.js

```diff
@@ -360,11 +379,10 @@ win32.format = function(pathObject) {
 
   var dir = pathObject.dir;
   var base = pathObject.base || '';
-  if (dir.slice(dir.length - 1, dir.length) === win32.sep) {
-    return dir + base;
-  }
-
   if (dir) {
+    if (dir[dir.length - 1] === win32.sep) {
```

Comment:
Does this change (moving the check inside `if (dir)`) fixes a bug where sometime `dir` would be undefined and thus the code would throw? If so, then it would be great to add at least a test for that.